dask_expr/_shuffle.py

```diff
@@ -1122,9 +1122,26 @@ def operation(df, index_name, drop, set_name, column_dtype):
         df.columns = df.columns.astype(column_dtype)
         return df
 
+    def _get_culled_divisions(self, divisions):
+        if self.frame.npartitions < len(divisions) - 1:
+            part_filter = list(self.frame.find_operations(PartitionsFiltered))
+            if len(part_filter) > 0:
+                return tuple(
+                    [
+                        div
+                        for i, div in enumerate(divisions)
+                        if i in part_filter[0]._partitions
+                    ]
+                    + [divisions[-1]]
```

Comment:
So this `part_filter` is always length==1 or do we only ever care about the first element in the list?

Reply:
We care about the first child that is filtered, you could imagine somtethong like this:


expr -> restrict to partitions 1-5 -> another expr -> another expr -> restrict to partitions 2 -> expr -> us

We only care about the filter to 2, the fact that 1-5 was filtered earlier is not relevant for us

Reply:
ah, I see. Thanks!